### ISSUE TITLE
[ROM/AST] AST initialization check in ROM fails on non-RMA LC states

### ISSUE THREAD

**Issue by milesdai** [lowRISC/opentitan#15728]
When the lifecycle state is set to anything other than RMA via the OTP, the [`ast_init_done`](https://github.com/lowRISC/opentitan/blob/4580a1765e7b5bd9a5f96c71f89e746e6c27c95d/sw/device/silicon_creator/lib/drivers/ast.c#L82) check fails.

To work around this temporarily, for tests that currently use non-RMA LC states in Bazel, we can apply the `otp_json_disable_ast_init` overlay which sets `CREATOR_SW_CFG_AST_INIT_EN` to `0x9`.

**Comment by msfschaffner** [comment]
CC @tjaychen what is the reason for this failure? should we check for `ast_init_done` if the `CREATOR_SW_CFG_AST_INIT_EN` is not set to True?

**Comment by tjaychen** [comment]
@milesdai sorry about this.
@msfschaffner and I were looking at something else and wandered by why you might be seeing this. 

Most of our otp defaults are in `otp_ctrl*.hjson`, but for some reason we left out `AST_INIT` as a default in the hjson and instead set that in the DV backdoor environments.

So this probably means when you guys migrated fully to this flow, without the default in there, your OTP was telling `mask_rom` to not initialize the AST, and this probably eventually led to the `ast_init_done` failrues.

although it's probably worth checking with @alphan , if the OTP doesn't say to initialize the AST, we probably shouldn't check the init status regardless of what LC state we are in.

**Comment by milesdai** [comment]
That makes sense. For now, I will set `CREATOR_SW_CFG_AST_INIT_EN` to 0x6 in the Bazel overlay to enable the AST initialization explicitly to bring the Bazel configuration in agreement with DV and close this issue. My understanding of the bug is that if the AST_INIT flag is set to enabled in OTP, then `rom_start.S` will initialize the AST. If the AST_INIT flag is disabled, then the C code skips the check for the initialization. However, because we were not explicitly enabling or disabling (i.e. leaving it a value of 0), then the assembly does not initialize the AST, but the C code does not skip the check.

We will eventually need to consolidate the configurations between DV and the FPGA, as discussed in #15337.

**Comment by alphan** [comment]
Discussed this offline, the complementary behavior of pre- and post-CRT code is intentional. post-CRT code also skips this check in TEST and RMA. The goal was to give us enough room to configure AST, but we will check and shutdown if it's not initialized unless it's explicitly disabled in OTP when the chip is in production life cycle states (dev, prod, prod_end).

**Comment by alphan** [comment]
> @alphan just to confirm, this means that during the manufacturing flow that we will have to set AST_INIT_EN "before" we set ROM_EXEC_EN right?

Not necessarily, but we need to set it before transitioning out of TEST* life cycle states.

**Comment by tjaychen** [comment]
ahhhhh, thank you for the reminder. I feel much better now. i keep forgetting we don't check certain things while in TEST states.
